### ISSUE TITLE
blocks and closures in method chains are visually indented

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#1329]
Which can cause orphans lower in the chain. E.g.,

```
    aaaaaaaaaaaaaaaa.map(|x| {
                            x += 1;
                            x
                        })
        .filter
```

The tricky thing here is that we do want to visually indent arguments to methods outside chains

**Comment by SamWhited** [comment]
Something very similar happens with multi-line macro invocations, although I'm not sure if this is the same issue so I'll paste it here and can move it to a new issue if not:

```rust
    TokenStream::from_str(quote!{
        #deserialized
    }
            .as_str())
        .unwrap()
```

[Playground](https://is.gd/Rz0BVO)

**Comment by topecongiro** [comment]
Closed via #1604.
